components/TextEditorContainer.tsx

```diff
@@ -0,0 +1,36 @@
+﻿import React, {MutableRefObject, useRef, useState} from "react";
+import Editor from "@monaco-editor/react";
+
+type EditorContentGetter = (
+    () => string
+);
```

Comment:
I kind of feel like it would be nicer to inline this and remove the outer parentheses. But maybe that's being picky. I'll let you decide what you want to do.

Reply:
I've made them inline